bindata/network/ovn-kubernetes-ipsec/ovn-ipsec.yaml

```diff
@@ -0,0 +1,207 @@
+      This DaemonSet launches the ovn ipsec networking components for all nodes.
+    release.openshift.io/version: "{{.ReleaseVersion}}"
+spec:
+  selector:
+    matchLabels:
+      app: ovn-ipsec
+  updateStrategy:
+    type: RollingUpdate
+  template:
+    metadata:
+      labels:
+        app: ovn-ipsec
+        component: network
+        type: infra
+        openshift.io/component: network
+        kubernetes.io/os: "linux"
+    spec:
+      serviceAccountName: ovn-kubernetes-node
+      hostNetwork: true
+      hostPID: true
+      priorityClassName: "system-node-critical"
+      # volumes in all containers:
+      # (container) -> (host)
+      # /etc/openvswitch -> /var/lib/openvswitch/etc - ovsdb system id
+      # /var/lib/openvswitch -> /var/lib/openvswitch/data - ovsdb data
+      # /run/openvswitch -> tmpfs - ovsdb sockets
+      # /env -> configmap env-overrides - debug overrides
+      containers:
+      # ovs-monitor-ipsec and libreswan daemons
+      - name: ovn-ipsec
+        image: "{{.OvnImage}}"
+        command:
+        - /bin/bash
+        - -c
+        - |
+          #!/bin/bash
+
+          function cleanup()
+          {
+            # Stop ovn-monitor-ipsec
+            /usr/share/openvswitch/scripts/ovs-ctl stop-ovs-ipsec
+            # Stop libreswan
+            /usr/libexec/ipsec/whack --shutdown
+            /sbin/ip xfrm policy flush
+            /sbin/ip xfrm state flush
+            /usr/sbin/ipsec --stopnflog
+            exit 0
+          }
+          trap cleanup SIGTERM
+
+          # Every time we restart the container, we will create a new key pair
+          echo "Configuring IPsec keys"
+
+          # We use the system-id as the CN. This is a requirement for OVN
+          cn=$(ovs-vsctl --retry -t 60 get Open_vSwitch . external-ids:system-id | tr -d "\"")
+
+          mkdir -p /etc/openvswitch/keys
+          pushd /etc/openvswitch/keys
+          umask 077 && openssl genrsa -out ipsec-privkey.pem 2048
+          openssl req -new -text \
+                  -extensions v3_req \
+                  -addext "subjectAltName = DNS:${cn}" \
+                  -subj "/C=US/O=ovnkubernetes/OU=kind/CN=${cn}" \
+                  -key ipsec-privkey.pem \
+                  -out ipsec-req.pem
+
+          csr_64=$(cat ipsec-req.pem | base64 | tr -d "\n")
+
+          # The signer controller does not allow re-signing a key. We will
+          # delete the old key to be sure it is not there
+          kubectl delete --ignore-not-found=true csr/$(hostname)
+
+          cat <<EOF | kubectl apply -f -
+          apiVersion: certificates.k8s.io/v1
+          kind: CertificateSigningRequest
+          metadata:
+            name: $(hostname)
+          spec:
+            request: ${csr_64}
+            signerName: network.openshift.io/signer
+            usages:
+            - ipsec tunnel
+          EOF
+
+          until [ ! -z $(kubectl get csr/$(hostname) -o jsonpath='{.status.certificate}' 2>/dev/null) ]
```

Comment:
Can't you just wait for the signed certificate to show up in the mount?

Reply:
I can't do this because the signed-cert does not appear via a secret. It is shared as a resource. This is ok as the signed cert is a public key that can be shared safely.

Reply:
ack.